rust/routee-compass-core/src/model/label/label_model.rs

```diff
@@ -0,0 +1,45 @@
+use allocative::Allocative;
+use serde::Serialize;
+
+use crate::model::{
+    label::label_model_error::LabelModelError,
+    network::VertexId,
+    state::{StateModel, StateVariable},
+};
+
+#[derive(PartialEq, Eq, Hash, Debug, Clone, Serialize, Allocative)]
+pub enum Label {
+    Vertex(VertexId),
+    VertexWithIntState {
+        vertex_id: VertexId,
+        state: Vec<u64>,
+    },
+}
+
+impl Label {
+    pub fn vertex_id(&self) -> VertexId {
+        match self {
+            Label::Vertex(vertex_id) => *vertex_id,
+            Label::VertexWithIntState { vertex_id, .. } => *vertex_id,
+        }
+    }
+}
+
+impl ToString for Label {
+    fn to_string(&self) -> String {
+        match self {
+            Label::Vertex(vertex_id) => format!("Vertex({})", vertex_id.0),
+            Label::VertexWithIntState { vertex_id, state } => {
+                format!("VertexWithIntState({}, {:?})", vertex_id.0, state)
+            }
+        }
+    }
+}
+pub trait LabelModel: Send + Sync {
+    fn label_from_state(
+        &self,
+        vertex_id: VertexId,
+        state: &[StateVariable],
+        state_model: &StateModel,
```

Comment:
Maybe we pass in the whole EdgeTraversal in place of VertexId and state here, which it contains along with the cost.

Reply:
I thought about doing that but it felt awkward when setting up the initial label and the edge traversal only has the edge id rather than the vertex id (unless I'm overlooking something)

Reply:
So, maybe what I mean (reading from my phone, probably missing some details here) is to pass in the `(VertexId, SearchTreeBranch)` tuple?